code/word_selection.py

```diff
@@ -0,0 +1,154 @@
+import time
+
+from talon import clip, Module, Context, actions
+
+mod = Module()
+
+
+@mod.action_class
+class Actions:
+    def word_neck(index: int):
+        """select the following word or the index'th word"""
+        word_neck(int(index))
+
+    def word_prev(index: int):
+        """select the previous word or the index'th word"""
+        word_prev(int(index))
+
+    def small_word_neck(index: int):
+        """select the following word or the index'th word"""
+        small_word_neck(int(index))
+
+    def small_word_prev(index: int):
+        """select the previous word or the index'th word"""
+        small_word_prev(int(index))
+
+    def big_word_neck(index: int):
+        """select the following word or the index'th word"""
+        big_word_neck(int(index))
+
+    def big_word_prev(index: int):
+        """select the previous word or the index'th word"""
+        big_word_prev(int(index))
+
+
+alphanumeric = "abcdefghijklmnopqrstuvwxyz0123456789_"
+
+
+def small_word_neck(index):
+    return word_neck(index, valid_characters=set(alphanumeric) - set("_"))
+
+
+def small_word_prev(index):
+    return word_prev(index, valid_characters=set(alphanumeric) - set("_"))
+
+
+def big_word_neck(index):
+    return word_neck(index, valid_characters=set(alphanumeric) | set("/\\-_.>=<"))
+
+
+def big_word_prev(index):
+    return word_prev(index, valid_characters=set(alphanumeric) | set("/\\-_.>=<"))
+
+
+def stop_selection(cursor_position):
+    assert cursor_position in ("left", "right")
+
+    with clip.capture() as s:
+        actions.key("shift-right")
+        time.sleep(0.25)
+        actions.edit.copy()
+    current_highlight = s.get()
+    actions.key("shift-left")
+
+    if len(current_highlight) > 1:
+        actions.key(cursor_position)
+
+
+def word_neck(word_index, valid_characters=alphanumeric):
+    old = clip.get()
+
+    stop_selection("right")
+
+    actions.key("shift-end")
+    time.sleep(0.25)
+    actions.key("cmd-c")
```

Comment:
This should probably use the platform/edit-agnostic actions

`edit.copy()`
`edit.left()`
`edit.right()`

What does "shift-end" do? I don't know if there's an action for this.

Reply:
shift-end selects until the end of the line; edit.extend_line_end()